Provider.php

```diff
@@ -69,11 +69,41 @@ public function getLastTouched(array $tableNamesOrIds)
                 array_merge($ids, $names)
             );
 
-            return $timestamp;
+            if ($lastTouched === null) {
+                return null;
+            }
+
+            $lastTouchedObject = new \DateTime($lastTouched);
+            return $lastTouchedObject->getTimestamp();
         }
     }
 
     /**
+     * Returns all tracked data rows and their respective last changes of a given table.
+     *
+     * @param string $tableName
+     * @return array
+     */
+    public function getLastTouchedOfEachRow($tableName)
+    {
+        $lastTouchedData = $this->connection->fetchAll('
+            SELECT m.data_id, m.last_touched
+            FROM wfd_meta m
+            JOIN wfd_table t on m.wfd_table_id = t.id
+            WHERE t.tablename = ?',
+            [$tableName]
+        );
+
+        $idAndVersionParis = [];
+        foreach ($lastTouchedData as $row) {
+            $lastTouchedObject = new \DateTime($row['last_touched']);
+            $idAndVersionParis[$row['data_id']] = $lastTouchedObject->getTimestamp();
```

Comment:
Now that I've seen this pattern at least three times I think a private helper method for date-string-to-timestamp conversion could be useful.